hyperjumptech/monika
# Detect timeout and uri not found

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. A lot of the errors previously are lumped at returned as a 500
2. This PR adds detection for timeut and not found
3. This PR closes #326 

## How did you implement / how did you fix it  
1.  Detect the events: "ECONNABORTED" and "ENOTFOUND" from axios

## How to test  
1. Run monika with some unknown uri and a quite timeout
2. Use the following config

  ```
  {
      "id": "httpbin1",
      "name": "probe httpbin",
      "requests": [
        {
          "method": "GET",
          "url": "https://xxxhttpbin.org/status/400",
          "saveBody": false,
          "timeout": 5000
        }
      ],
      "incidentThreshold": 3,
      "recoveryThreshold": 3
    },
    {
      "id": "httpbin2",
      "name": "probe httpbin",
      "requests": [
        {
          "method": "GET",
          "url": "https://httpbin.org/status/400",
          "saveBody": false,
          "timeout": 50
        }
      ],
      "incidentThreshold": 3,
      "recoveryThreshold": 3
    }
```
3. you should get something like:
```
WARN: 2021-08-30T03:31:33.680Z 1 id:httpbin1 0 GET https://xxxhttpbin.org/status/400 78ms, ALERT: URI not found, status-not-2xx
WARN: 2021-08-30T03:31:33.683Z 1 id:httpbin2 599 GET https://httpbin.org/status/400 53ms, ALERT: Request Timed out, status-not-2xx
```
